### ISSUE TITLE
Explain autograd, gradient mathematically

### ISSUE THREAD

**Issue by vuhung16au** [vuhung16au/pytorch-mastery#54]
Explain autograd, gradient mathematically using the example below.

Save your response to 

`pytorch-mastery/examples/pytorch-tutorials/03_pytorch_autograd-explain.md`

Pls use LaTex.

Let's say, 

we have x = sydney_revenue

express `seasonal_boost` as a function s(x)
express `weekend_bonus` as a function of `seasonal_boost`
express `after_tax` as a function of weekend_bonus`

then 
- calculate the revenue
- calculate the gradient 

```
# Basic autograd demonstration
print("🇦🇺 Sydney Tourism Revenue Analysis with Autograd\n")

# Create tensor with gradient tracking
sydney_revenue = torch.tensor([150.0], requires_grad=True)
print(f"Initial revenue: ${sydney_revenue.item():.0f}k AUD")
print(f"Requires grad: {sydney_revenue.requires_grad}")

# Build computation graph
seasonal_boost = sydney_revenue * 1.2  # +20% summer boost
weekend_bonus = seasonal_boost + 50.0  # +$50k weekend
after_tax = weekend_bonus * 0.9       # -10% tax

print(f"\nRevenue calculation:")
print(f"Base: ${sydney_revenue.item():.0f}k")
print(f"After boost: ${seasonal_boost.item():.0f}k")
print(f"After bonus: ${weekend_bonus.item():.0f}k")
print(f"After tax: ${after_tax.item():.0f}k")

# Check computation graph
print(f"\nComputation graph:")
print(f"seasonal_boost.grad_fn: {seasonal_boost.grad_fn}")
print(f"weekend_bonus.grad_fn: {weekend_bonus.grad_fn}")
print(f"after_tax.grad_fn: {after_tax.grad_fn}")

# Compute gradients
after_tax.backward()
gradient = sydney_revenue.grad.item()

print(f"\nGradient: {gradient:.2f}")
print(f"Interpretation: $1k increase in base revenue → ${gradient:.2f}k increase in final revenue")

# Manual verification: 1.2 * 0.9 = 1.08
expected = 1.2 * 0.9
print(f"Expected: {expected:.2f} ✓")
```
